include/gsl/gsl-lite.hpp

```diff
@@ -4572,7 +4572,7 @@ struct hash< ::gsl::not_null< T > > : public ::gsl::detail::conditionally_enable
 {
 public:
     gsl_NODISCARD gsl_constexpr std::size_t
-    operator()( const ::gsl::not_null< T >& v ) const gsl_noexcept_iff( hash<T>()( declval<T>() ) )
+    operator()( const ::gsl::not_null< T >& v ) const gsl_noexcept_if(gsl_noexcept( hash<T>()(declval<T>() ) ) )
```

Comment:
...so this statement is expanded to `noexcept(noexcept(noexcept(...)))`, which is always `true`.